apps/studio/src/lib/db/clients/clickhouse/ClickHouseCursor.ts

```diff
@@ -0,0 +1,60 @@
+import { BeeCursor, OrderBy, TableFilter } from "../../models"
+import rawlog from 'electron-log'
+import { ClickHouseClient as ClickHouse } from "../clickhouse"
+import type { ClickHouseClient } from '@clickhouse/client'
+import { uuidv4 } from "@/lib/uuid"
+const log = rawlog.scope('clickhouse/cursor')
+
+
+interface CursorOptions {
+  table: string
+  orderBy: OrderBy[]
+  filters: string | TableFilter[]
+  client: ClickHouseClient
+  chunkSize: number
+}
+
+export class ClickHouseCursor extends BeeCursor {
+  private readonly options: CursorOptions
+  private cursorPos = 0;
+  private client: ClickHouseClient;
+  private queryId: string;
+
+  constructor(options: CursorOptions) {
+    super(options.chunkSize)
+    this.options = options
+    this.client = options.client
+    this.queryId = uuidv4();
+  }
+
+  async start() {
+    log.info("Starting cursor");
+  }
+
+  async read() {
+    const offset = this.cursorPos * this.chunkSize;
+    const limit = this.chunkSize;
+    const query = ClickHouse.buildSelectTopQuery(
+      this.options.table,
+      offset,
+      limit,
+      this.options.orderBy,
+      this.options.filters
+    );
+    const result = await this.client.query({
```

Comment:
This isn't running inside of a transaction, so this cursor isn't really generating a consistent view right? Are there no streaming APIs for node-clickhouse?

Also according to the docs, the query_id must be different on each call:
```
If you are overriding the query_id parameter, you need to ensure its uniqueness for **every call**. A random UUID is a good choice.
```

Reply:
Their github has some examples of using streams. I recommend using `JSONCOMPACTEACHROW` -https://clickhouse.com/docs/en/interfaces/formats#jsoncompacteachrow

Original discussion:
https://github.com/ClickHouse/clickhouse-js/issues/243

Example code:
https://github.com/ClickHouse/clickhouse-js/blob/a351508eda15c27e47ffcaa4ec96b63016d2e87b/examples/node/select_streaming_for_await.ts

Reply:
There's a similar code in firebird cursor so I'll fix that as well 😅